fe/fe-core/src/main/java/org/apache/doris/analysis/PrepareStmt.java

```diff
@@ -163,13 +164,15 @@ public void analyze(Analyzer analyzer) throws UserException {
         if (inner instanceof SelectStmt) {
             // Use tmpAnalyzer since selectStmt will be reAnalyzed
             Analyzer tmpAnalyzer = new Analyzer(context.getEnv(), context);
+            tmpAnalyzer.setPrepareStmt(true);
             SelectStmt selectStmt = (SelectStmt) inner;
             inner.analyze(tmpAnalyzer);
-            if (!selectStmt.checkAndSetPointQuery()) {
-                throw new UserException("Only support prepare SelectStmt point query now");
+            isPointQuery = selectStmt.checkAndSetPointQuery();
+            if (isPointQuery) {
```

Comment:
statment should be these types `NONE, FULL_PREPARED, STATEMENT`, STATEMENT type should suite. no need to `setPrepareStmt`, use type directly